tools/eclipse_prj_helper/generate_prj.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+###########################################################################
+#
+#  Copyright (c) 2013-2017, ARM Limited, All Rights Reserved
+#  SPDX-License-Identifier: Apache-2.0
+#
+#  Licensed under the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+###########################################################################
+import argparse
+import os
+import subprocess
+import string
+import xml.etree.ElementTree as ElementTree
+
+
+script_dir = os.path.dirname(os.path.abspath(__file__))
+
+
+def get_parser():
+
+    parser = argparse.ArgumentParser(
+        description='Eclipse project generator',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+
+    parser.add_argument(
+        '-w',
+        '--workspace',
+        required=True,
+        help='Application/Example root directory for hosting Eclipse project files'
+    )
+
+    parser.add_argument(
+        '-m',
+        '--target',
+        default='K64F',
+        help='Compile target MCU. Example: K64F, NUCLEO_F401RE,NRF51822...'
+    )
+    parser.add_argument(
+        '-t',
+        '--toolchain',
+        default='GCC_ARM',
+        help='Compile toolchain. Example: ARM, GCC_ARM, IAR'
+    )
+
+    parser.add_argument(
+        '-i',
+        '--ide',
+        default='eclipse_gcc_arm',
+        help='IDE to create project files for. Example: UVISION4, UVISION5, GCC_ARM, IAR, COIDE'
+    )
+
+    return parser
+
+
+def patch_makefile(workspace, target, toolchain, uvisor_dir, importer_dir):
+    template_file = os.path.join(script_dir, 'Makefile.template')
+    with open(template_file, 'rt') as fh:
+        makefile_template = string.Template(fh.read())
+
+    # overwrite generated Makefile
+    with open(os.path.join(workspace, 'Makefile'), 'wt') as fh:
+        fh.write(
+            makefile_template.safe_substitute(
+                workspace_dir=workspace,
+                uvisor_dir=uvisor_dir,
+                importer_dir=importer_dir,
+                target=target,
+                toolchain=toolchain,
+                template_file=template_file
+            )
+        )
```

Comment:
@theotherjimmy - there is always more than one way for doing stuff :)
Personally I do prefer the "right tool for the right job" approach. Jinja seems to be an overkill for such a simple task - since this one is a simple, straight forward string substitution. There is no complex logic involved requiring conditional statements and/or loops.
BTW, Template.safe_substitute does rais KeyError exception in case placeholders are missing.